stories/icicle/02_unix_flame.tsx

```diff
@@ -0,0 +1,116 @@
+/*
+ * Licensed to Elasticsearch B.V. under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch B.V. licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+import React from 'react';
+
+import { Chart, Datum, Partition, PartitionLayout, Settings } from '../../src';
+import { ShapeTreeNode } from '../../src/chart_types/partition_chart/layout/types/viewmodel_types';
+import { PrimitiveValue } from '../../src/chart_types/partition_chart/layout/utils/group_by_rollup';
+import { mocks } from '../../src/mocks/hierarchical';
+import { STORYBOOK_LIGHT_THEME } from '../shared';
+import { categoricalFillColor, plasma18 as palette } from '../utils/utils';
+
+const color = palette.slice().reverse();
+
+const raw = mocks.observabilityTree;
+
+interface Node {
+  c?: Node[];
+  n: string;
+  v: number;
+}
+
+type Row = { [layerKey: string]: unknown; value: number; depth: number };
+
+const flatTree = ({ c, n, v }: Node, depth: number): Row[] => {
+  if (!c) {
+    return [{ [`layer_${depth}`]: n, value: v, depth }];
+  }
+  // looks like our test runner can't run c.flatMap(...)
+  // eslint-disable-next-line @typescript-eslint/ban-ts-comment
+  // @ts-ignore
+  // eslint-disable-next-line prefer-spread
+  const childrenRows: Row[] = [].concat.apply(
+    [],
+    // eslint-disable-next-line @typescript-eslint/ban-ts-comment
+    // @ts-ignore
+    c.map((child) => flatTree(child, depth + 1)),
+  );
```

Comment:
How is this different than this?

```suggestion
  const childrenRows = c.reduce<Row[]>((a, child) => [...a, ...flatTree(child, depth + 1)], []);
```

> Then there's no eslint comments needed.

Reply:
We should figure out why `flatMap` is not working too cuz that would be very helpful.

Reply:
Your version looks better in all regards; l'll create an issue for no `flatMap` in the runner